test/unit/dinero.spec.js

```diff
@@ -247,8 +247,11 @@ describe('Dinero', () => {
       expect(shares[0].getAmount()).toBe(25)
       expect(shares[1].getAmount()).toBe(75)
     })
-    test('should throw when one of the ratios equals to zero', () => {
-      expect(() => Dinero({ amount: 1003 }).allocate([60, 0, 10, 30])).toThrow()
+    test('should allocate the amount of the Dinero object into new ones and distribute the remainder when one of the ratios equals to zero', () => {
+      const shares = Dinero({ amount: 1003 }).allocate([50, 0, 50])
+      expect(shares[0].getAmount()).toBe(502)
+      expect(shares[1].getAmount()).toBe(0)
+      expect(shares[2].getAmount()).toBe(501)
```

Comment:
This use case works because you're splitting amounts in 2 (50/50) and because the first ratio is > 0. The remainder is 1, therefore it's distributed to the first ratio and it stops there.

If you do `Dinero({ amount: 1003 }).allocate([0, 50, 50])`, this will fail because the remainder will distribute to the first ratio (0) without verifying if it should.

I'd recommend adding that test case (or using the one that you deleted) and to adapt the algorithm of the `allocate` method to make it pass.

Additionally, I realized that the way I used to assert for `allocate` (several assertions in a single test) isn't optimal, because Jest exits on the very first breaking test and doesn't show you the whole distributed ratios.

Something like this would work better IMHO (I'll refactor the other tests in another PR):

```js
expect(shares.map(share => share.getAmount())).toEqual([0, 502, 501])
```

Jest feedback:

![Capture d’écran 2020-03-07 à 13 32 00](https://user-images.githubusercontent.com/5370675/76143491-2b7a5180-6078-11ea-8385-e5c98cee8ad6.png)

Reply:
Ouch. I realized that the better way to allocate remainder would be allocate it to ordered ratios, but it will be breaking change. Right now I've saved remainder allocation order.

I've added the test to illustrate my idea.
```js
const shares = Dinero({ amount: 1 }).allocate([0, 1, 50])
expect(shares.map(share => share.getAmount())).toEqual([0, 1, 0])
```

I think in the `2.0.0` version the order of the result items should be changed.

Thank you.

Reply:
Additionally, could you please fix this line:
https://github.com/dinerojs/dinero.js/blob/develop/.all-contributorsrc#L3

When I run `npm run contributors:generate` all contributor tags was changed to `https://github.com/sarahdayan/dinero.js`